ci/deps.sh

```diff
@@ -34,40 +35,72 @@ echo "MAKEFLAGS: ${MAKEFLAGS}"
 mkdir -p /tmp
 cd /tmp
 
+gh_retrieve_wget(){
+    ORG=$1
+    REPO=$2
+    TAG=$3
+
+    echo "Downloading $ORG $REPO $TAG"
+    
+
+
+    wget $WGET_FLAGS https://github.com/$ORG/$REPO/archive/refs/tags/$TAG.tar.gz && \
+    mkdir $REPO && \
+    tar xzfv $TAG.tar.gz --strip-components=1 -C $REPO && \
+    rm $TAG.tar.gz
+}
+
+gh_retrieve_git(){
+    ORG=$1
+    REPO=$2
+    TAG=$3
+
+    echo "Downloading $ORG $REPO $TAG"
+    git clone --branch $TAG --recursive https://www.github.com/$ORG/$REPO
+}
+
+gh_build(){
+    ORG=$1
+    REPO=$2
+    TAG=$3
```

Comment:
What if the script supported a default value for the tag (like master/main)? That would require reordering the args; again, up to the project owners, but maintaining version pinning requires focus, and the chance of shipping software with (even security) known bugs can be a problem... One approach I have been using in projects where we don't have the time to timely push versions of our deps is "pull latest versions of our deps and rely on continuous testing to identify breaking changes the sooner the better; fallback to specific pins if we detect a breaking change and we cannot immediately adapt our code", but not saying this approach is the right one for this project... a pity we dont have a proper deps resolver that let's us to cleanly implement hybrid models (for A pick the latest; for B pick any 1.x; for C pick 2.1.4).